services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplUnitTest.java

```diff
@@ -899,29 +900,115 @@ public void testGetUserAuthorization() {
+		
+		// method under test
+		openIDConnectManagerImpl.getUserAuthorization(token);
+		
+		verify(mockOauthClientDao, never()).getSectorIdentifierSecretForClient(OAUTH_CLIENT_ID);
+		verify(mockOauthClientDao, never()).isOauthClientVerified(AuthorizationConstants.SYNAPSE_OAUTH_CLIENT_ID);
+	}
+
+	@Test
+	public void testGetAdminUserAuthorizationSynapseOAuthClient() {
+		String token = "access token";
+		when(oidcTokenHelper.parseJWT(token)).thenReturn(mockJWT);
+		Claims claims = Jwts.claims();
+		List<OAuthScope> scopes = Arrays.asList(OAuthScope.values());
+		Map<OIDCClaimName, OIDCClaimsRequestDetails> oidcClaims = Collections.singletonMap(OIDCClaimName.email, null);
+		ClaimsJsonUtil.addAccessClaims(scopes, oidcClaims, claims);
+		when(mockJWT.getBody()).thenReturn(claims);
+		claims.setAudience(AuthorizationConstants.SYNAPSE_OAUTH_CLIENT_ID);		
+		claims.setSubject(USER_ID);	
+		Set<Long> adminUsersGroups = ImmutableSet.of(USER_ID_LONG, 
+				AuthorizationConstants.BOOTSTRAP_PRINCIPAL.ADMINISTRATORS_GROUP.getPrincipalId());
+		when(mockUserManager.getUserGroups(USER_ID_LONG)).thenReturn(adminUsersGroups);
+		
+		// method under test
+		UserInfo actual = openIDConnectManagerImpl.getUserAuthorization(token);
+		
+		assertTrue(actual.isAdmin());		
+	}
+
+	@Test
+	public void testGetAdminUserAuthorizationSynapseOAuthClientNotFullScope() {
+		String token = "access token";
+		when(oidcTokenHelper.parseJWT(token)).thenReturn(mockJWT);
+		Claims claims = Jwts.claims();
+		List<OAuthScope> scopes = ImmutableList.of(OAuthScope.openid, OAuthScope.view);
+		Map<OIDCClaimName, OIDCClaimsRequestDetails> oidcClaims = Collections.singletonMap(OIDCClaimName.email, null);
+		ClaimsJsonUtil.addAccessClaims(scopes, oidcClaims, claims);
+		when(mockJWT.getBody()).thenReturn(claims);
+		claims.setAudience(AuthorizationConstants.SYNAPSE_OAUTH_CLIENT_ID);		
+		claims.setSubject(USER_ID);	
+		Set<Long> adminUsersGroups = ImmutableSet.of(USER_ID_LONG, 
+				AuthorizationConstants.BOOTSTRAP_PRINCIPAL.ADMINISTRATORS_GROUP.getPrincipalId());
+		when(mockUserManager.getUserGroups(USER_ID_LONG)).thenReturn(adminUsersGroups);
+		
+		// method under test
+		UserInfo actual = openIDConnectManagerImpl.getUserAuthorization(token);
+		
+		assertFalse(actual.isAdmin());		
+	}
+
+	@Test
+	public void testGetAdminUserAuthorizationSynapseOAuthClientNotAdminGroup() {
+		String token = "access token";
+		when(oidcTokenHelper.parseJWT(token)).thenReturn(mockJWT);
+		Claims claims = Jwts.claims();
+		List<OAuthScope> scopes = Arrays.asList(OAuthScope.values());
+		Map<OIDCClaimName, OIDCClaimsRequestDetails> oidcClaims = Collections.singletonMap(OIDCClaimName.email, null);
+		ClaimsJsonUtil.addAccessClaims(scopes, oidcClaims, claims);
+		when(mockJWT.getBody()).thenReturn(claims);
+		claims.setAudience(AuthorizationConstants.SYNAPSE_OAUTH_CLIENT_ID);		
+		claims.setSubject(USER_ID);	
+		Set<Long> adminUsersGroups = ImmutableSet.of(USER_ID_LONG);
```

Comment:
maybe rename to just userGroups since this is really not admin groups

Reply:
Done.